students/JerryH/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# List of sample donors and donations
+donors = {
+    "Bill Gates": [234.22, 45645.24, 43953.09, 98823],
+    'Jeff Bezo': [4564.23],
+    'Mike Dell': [299.09, 26273.67],
+    'Harry Potter': [8234.09, 83948.04, 7834.23],
+    'Ben Williams': [29283.00, 1334.34],
+    'Guy James': [93.00, 34.34]
+    }
+
+options = {
+    'THANK_YOU_OPT': 1,
+    'REPORT_OPT': 2,
+    'SEND_LETTER_OPT': 3,
+    'QUIT_OPT': 4
+    }
```

Comment:
as part of the instructions we have: "See if you can use a dict to switch between the users selections. "

it looks like you tried to incorporate a dict into your options.. but you can take it a step further and do something like:

```
selection_map = {
    "1": send_thank_you,
    "2": create_report,
    "3": send_letters,
    "4": quit_program
}
```
the dict values are function objects, and it will act like a switch based on user input for the option.
you use will use it like this:
```
option_value = input("..")
selection_map[optiona_value]()
```

you can also use `selection_map` for validation:
```
if option_value not in selection_map.values() :
     # continue looping asking for user input
```

above pattern will also encourage you to break up code into more functional blocks (by using functions)